R/makeTreeSummarizedExperimentFromqiime2.R

```diff
@@ -0,0 +1,245 @@
+#'   `BIOMV210DirFmt` (feature table), `TSVTaxonomyDirectoryFormat` (taxonomic
+#'   table), `NewickDirectoryFormat` (phylogenetic tree ) and
+#'   `DNASequencesDirectoryFormat` (representative sequences) are supported
+#'    right now.
+#' @param temp character, a temporary directory in which the qza file will be
+#'   decompressed to, default `tempdir()`.
+#' @return [`phyloseq::otu_table-class`] object for feature table,
+#'   [`phyloseq::taxonomyTable-class`] object for taxonomic table,
+#'   [`ape::phylo`] object for phylogenetic tree,
+#'   [`Biostrings::DNAStringSet-class`] for representative sequences of taxa.
+#' @noRd
+#' @keywords internal
+read_qza <- function(file, temp = tempdir()) {
+    unzipped_file <- utils::unzip(file, exdir = temp)
+    meta_file <- grep("metadata.yaml", unzipped_file, value = TRUE)
+    metadata <- yaml::read_yaml(meta_file[1])
+    format <- metadata$format
+    uuid <- metadata$uuid
+
+    if (grepl("BIOMV", metadata$format)) {
+        biom_file <- file.path(temp, uuid,"data/feature-table.biom")
+        res <- read_q2biom(biom_file)
+    } else if (format == "TSVTaxonomyDirectoryFormat") {
+        taxa_file <- file.path(temp, uuid, "data/taxonomy.tsv")
+        res <- read_q2taxa(taxa_file)
+    } else if (format == "NewickDirectoryFormat") {
+        tree_file <- file.path(temp, uuid, "data/tree.nwk")
+        res <- ape::read.tree(tree_file)
+    } else if (format == "DNASequencesDirectoryFormat") {
+        refseq_file <- file.path(temp, uuid, "data/dna-sequences.fasta")
+        res <- Biostrings::readDNAStringSet(refseq_file)
+    } else {
+        stop(
+            "Only files in format of 'BIOMV210DirFmt', ",
+            "'TSVTaxonomyDirectoryFormat', NewickDirectoryFormat' and ",
+            "'DNASequencesDirectoryFormat' are supported."
+        )
+    }
+
+    res
+}
+
+
+#' Read qiime2 feature table
+#'
+#' Import qiime2 feature table into otu_table object
+#'
+#' @param file character, file name of the biom file.
+#' @return [`phyloseq::otu_table-class`] object.
+#' @noRd
+#' @keywords internal
+read_q2biom <- function(file) {
+    biomobj <- suppressWarnings(biomformat::read_biom(file))
+    feature_tab <- as(biomformat::biom_data(biomobj),"matrix")
+
+    feature_tab
+}
+
+#' Read qiime2 taxa file
+#' @keywords internal
+#' @noRd
+read_q2taxa <- function(file) {
+    taxa <- utils::read.table(file, sep = '\t', header = TRUE)
+    if ("Confidence" %in% names(taxa)) {
+        taxa$Confidence <- NULL
+    }
+
+    as.matrix(taxa)
+}
+
+#' Read qiime2 sample meta data file
+#' @keywords internal
+#' @noRd
+read_q2sample_meta <- function(file) {
+    sam <- utils::read.table(file = file, header = TRUE, sep = "\t", comment.char = "")
+    rownames(sam) <- as.character(sam[, 1])
+
+    # remove row: #q2:types
+    idx <- which(sam == "#q2:types", arr.ind = TRUE)
+    sam <- sam[-idx[, "row"], ]
+
+    S4Vectors::DataFrame(sam)
+}
```

Comment:
See above regarding internal functions

Solve imports

Why is the confidence deleted? `TSE` can hold all data in row `rowData`, not just taxonomic